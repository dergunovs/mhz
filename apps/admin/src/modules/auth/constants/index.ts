export const API_POST_LOGIN = 'auth/login';
export const API_GET_CHECK_AUTH = 'auth/check';

export const URL_LOGIN = '/';
