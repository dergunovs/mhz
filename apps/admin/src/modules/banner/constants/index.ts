export const URL_BANNER = '/banners';
export const URL_BANNER_CREATE = `${URL_BANNER}/create`;
export const URL_BANNER_EDIT = `${URL_BANNER}/edit`;
