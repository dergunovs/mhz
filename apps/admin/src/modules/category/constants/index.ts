export const URL_CATEGORY = '/categories';
export const URL_CATEGORY_CREATE = `${URL_CATEGORY}/create`;
export const URL_CATEGORY_EDIT = `${URL_CATEGORY}/edit`;

export const CATEGORY_FIELD_TYPE_OPTIONS = ['string', 'number', 'boolean'];
