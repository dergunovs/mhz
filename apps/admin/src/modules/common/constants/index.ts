export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
