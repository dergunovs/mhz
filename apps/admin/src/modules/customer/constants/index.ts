export const URL_CUSTOMER = '/customers';

export const API_CUSTOMER = 'customer';
