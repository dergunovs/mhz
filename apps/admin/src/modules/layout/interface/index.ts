import { FunctionalComponent } from 'vue';

export interface INavItem {
  _id: string;
  url: string;
  title: string;
  icon: FunctionalComponent;
}

export interface IPageTitleLink {
  url: string;
  title?: string;
}
