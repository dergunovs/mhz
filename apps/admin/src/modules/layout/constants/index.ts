import { INavItem } from '@/layout/interface';

export const NAV_ITEMS: INavItem[] = [
  { _id: 0, url: '/', title: 'Главная' },
  { _id: 1, url: '/users', title: 'Пользователи' },
  { _id: 2, url: '/shipments', title: 'Доставка' },
  { _id: 3, url: '/orders', title: 'Заказы' },
  { _id: 4, url: '/categories', title: 'Категории' },
  { _id: 5, url: '/products', title: 'Товары' },
  { _id: 5, url: '/manufacturers', title: 'Производители' },
];
