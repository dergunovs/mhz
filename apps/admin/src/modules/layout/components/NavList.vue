<template>
  <nav :class="$style.nav">
    <NavItem v-for="item in NAV_ITEMS" :key="item._id" :navItem="item" data-test="nav-list-item" />
  </nav>
</template>

<script setup lang="ts">
import NavItem from '@/layout/components/NavItem.vue';

import { NAV_ITEMS } from '@/layout/constants';
</script>

<style module lang="scss">
.nav {
  display: flex;
  flex-direction: column;
  gap: 8px;
  width: 216px;
  height: calc(100dvh - 64px);
  padding: 32px 16px;
  overflow-y: auto;
}
</style>
