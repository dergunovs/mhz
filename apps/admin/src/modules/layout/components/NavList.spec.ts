import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import NavList from './NavList.vue';

import { wrapperFactory } from '@/common/test';
import { NAV_ITEMS } from '@/layout/constants';

const navListItem = dataTest('nav-list-item');

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(NavList, {});
});

enableAutoUnmount(afterEach);

describe('NavList', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(NavList)).toBeTruthy();
  });

  it('shows nav list items', async () => {
    expect(wrapper.findAllComponents(navListItem).length).toEqual(NAV_ITEMS.length);
  });
});
