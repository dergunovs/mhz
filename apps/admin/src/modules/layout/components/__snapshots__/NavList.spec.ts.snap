// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavList > matches snapshot 1`] = `
"<nav class="_nav_1901ef">
  <nav-item-stub navitem="[object Object]" data-test="nav-list-item"></nav-item-stub>
  <nav-item-stub navitem="[object Object]" data-test="nav-list-item"></nav-item-stub>
  <nav-item-stub navitem="[object Object]" data-test="nav-list-item"></nav-item-stub>
  <nav-item-stub navitem="[object Object]" data-test="nav-list-item"></nav-item-stub>
  <nav-item-stub navitem="[object Object]" data-test="nav-list-item"></nav-item-stub>
  <nav-item-stub navitem="[object Object]" data-test="nav-list-item"></nav-item-stub>
  <nav-item-stub navitem="[object Object]" data-test="nav-list-item"></nav-item-stub>
  <nav-item-stub navitem="[object Object]" data-test="nav-list-item"></nav-item-stub>
</nav>"
`;
